qutebrowser/misc/ipc.py

```diff
@@ -322,6 +323,13 @@ def on_ready_read(self):
                 return
 
             try:
+                target_arg = json_data['target_arg']
+            except KeyError:
+                log.ipc.error("target arg missing: {}".format(decoded.strip()))
+                self._handle_invalid_data()
+                return
```

Comment:
I think it'd be cleaner to do something like:

```python
for name in ('args', 'target_arg'):
    if name not in json_data:
        log.ipc.error("Missing {}: {}".format(name, decoded.strip()))
        self._handle_invalid_data()
        return
```

and then use `json_data['args']` and `json_data['target_arg']` in the `self.got_args.emit` line. Then you can also get rid of the `pylint: disable=too-many-return-statements` again.